CMakeLists.txt

```diff
@@ -181,6 +181,33 @@ endif()
 
 if(NOT Protobuf_FOUND)
   find_package(Protobuf REQUIRED)
+
+  # orchestrated effort of both protobuf & cmake are responsible for the next lines..
+
+  # TODO: monitor how protobuf Linux packages evolve. Eventually everyone should use
+  # the Apple configuration, but e.g. apt does not install CMake config with protobuf
+  if(Protobuf_VERSION VERSION_GREATER_EQUAL 4.22.0)
+    add_definitions(-DPROTOBUF_USE_DLLS)
+    if(APPLE OR WIN32)
+      # CONFIG mode is ideal for new abseil protobuf builds, which uses the protobuf-provided
+      # CMake configuration instead of CMake internal one, as that's not looking for abseil:
+      # https://github.com/valhalla/valhalla/issues/4173#issuecomment-1638067950
+      # Newer protobuf versions require a protobuf_MODULE_COMPATIBLE to be able 
+      # to see the protobuf_generate_cpp() CMake function
+      set(protobuf_MODULE_COMPATIBLE ON CACHE BOOL "")
+      find_package(Protobuf REQUIRED CONFIG)
+    else()
+      # linux might not have CMake config installed (e.g. Debian derivatives)
+      # need to link protobuf to abseil manually
+      find_package(Protobuf REQUIRED)
+      find_package(absl REQUIRED)
+      if(TARGET protobuf::libprotobuf-lite)
+        target_link_libraries(protobuf::libprotobuf-lite INTERFACE "absl::absl_check;absl::absl_log;absl::algorithm;absl::base;absl::bind_front;absl::bits;absl::btree;absl::cleanup;absl::cord;absl::core_headers;absl::debugging;absl::die_if_null;absl::dynamic_annotations;absl::flags;absl::flat_hash_map;absl::flat_hash_set;absl::function_ref;absl::hash;absl::layout;absl::log_initialize;absl::log_severity;absl::memory;absl::node_hash_map;absl::node_hash_set;absl::optional;absl::span;absl::status;absl::statusor;absl::strings;absl::synchronization;absl::time;absl::type_traits;absl::utility;absl::variant")
```

Comment:
I took this list from my local protobuf config cmake file, it's how abseil is linked there. I'm sure we hardly need all of them, but I felt no desire singling them out as it pretty is irrelevant

Reply:
I should add that this is not necessarily super safe to use, that's the configuration for the exact version of protobuf that I have installed. I can't really tell if this also works for smth between 3.22.0 and my 3.23.4, and of course this might break for future protobuf versions. It might result for a longer while in a series of version ifs. It'd be nice to make valhalla compile again without local patches (which tend to get committed and pushed). I'm not sure what else to try.

@rochad can you try this branch if it gets rid of your build problems on OSX?

Reply:
we can make it link just the ones that protobuf actually needs. like delete all of them and add them back one at a time until it doesnt have linker errors anymore. seems like to start with you might only need: `absl::absl_log`